apps/tangle-dapp/src/data/restake/useNativeRestakeTx.ts

```diff
@@ -0,0 +1,66 @@
+import { BN } from '@polkadot/util';
+import { useCallback } from 'react';
+
+import { TxName } from '../../constants';
+import { PrecompileAddress } from '../../constants/evmPrecompiles';
+import useAgnosticTx from '../../hooks/useAgnosticTx';
+import { EvmTxFactory } from '../../hooks/useEvmPrecompileCall';
+import useFormatNativeTokenAmount from '../../hooks/useFormatNativeTokenAmount';
+import { SubstrateTxFactory } from '../../hooks/useSubstrateTx';
+import { GetSuccessMessageFn } from '../../types';
+import RESTAKING_PRECOMPILE_ABI from '../../abi/restaking';
+import { SubstrateAddress } from '@tangle-network/ui-components/types/address';
+import { convertAddressToBytes32 } from '@tangle-network/ui-components';
+
+type Context = {
+  amount: BN;
+  operatorAddress: SubstrateAddress;
+};
+
+const useNativeRestakeTx = () => {
+  const formatNativeTokenAmount = useFormatNativeTokenAmount();
+
+  const evmTxFactory: EvmTxFactory<
+    typeof RESTAKING_PRECOMPILE_ABI,
+    'delegateNomination',
+    Context
+  > = useCallback(
+    (context) => ({
+      functionName: 'delegateNomination',
+      arguments: [
+        convertAddressToBytes32(context.operatorAddress),
+        BigInt(context.amount.toString()),
+        // TODO: Blueprint selection.
```

Comment:
can we get blueprint selection added? I believe @monaiuu designed the component btw.